### PR TITLE
Setup UV and .gitignore

### PR DESCRIPTION
**Pull request by TheEnquirer**
Let's use `uv`! It's faster and a way better dependency and build manager.

Also added a `.gitignore` for future dev

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[project]
+name = "embryo-ode"
```
Comment (michel-mata) [inline pyproject.toml:6]:
```suggestion
name = "embryo"
```

Comment (michel-mata) [reply pyproject.toml:6]: For packaging, we should avoid the dash in the name of our project.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+### Python Patch ###
+# Poetry local configuration file - https://python-poetry.org/docs/configuration/#local-configuration
+poetry.toml
+
+# ruff
+.ruff_cache/
+
+# LSP config files
+pyrightconfig.json
```
Comment (michel-mata) [inline .gitignore:171]:
```suggestion
pyrightconfig.json

# OSX
.DS_Store
```

**Review by michel-mata — COMMENTED** [review]
Thanks for all the dev setup!

File: pyproject.toml (hunk 1 above)
Comment (denisfaer) [inline pyproject.toml:6]:
```suggestion
name = "epipeODE"
```

Comment (denisfaer) [reply pyproject.toml:6]: If we are to have this as a package the name should be very specific as we are only modeling one step of embryogenesis: hence epipeODE.

**Pull request merged**